### ISSUE TITLE
System test testNodeDiscoveryRest fails on OS X using docker-machine

### ISSUE THREAD

**Issue by jhftrifork** [mesos/elasticsearch#200]
## Steps to reproduce

Use OS X? (Bug has only appeared on OS X.)

Run commands from the README:

```
$ docker-machine create -d virtualbox --virtualbox-memory 4096 --virtualbox-cpu-count 2 mesos-es
$ eval $(docker-machine env mesos-es)
$ ./gradlew build buildDockerImage system-test:systemTest
```

## Expected behavior

?

## Actual behavior

```
org.apache.mesos.elasticsearch.systemtest.DiscoverySystemTest > testNodeDiscoveryRest STANDARD_OUT
    [INFO] 2015-07-31 11:17:19,091 org.apache.mesos.elasticsearch.systemtest.DiscoverySystemTest call - Polling Elasticsearch _nodes endpoints...
    [INFO] 2015-07-31 11:17:20,122 org.apache.mesos.elasticsearch.systemtest.DiscoverySystemTest call - Polling Elasticsearch _nodes endpoints...
    [INFO] 2015-07-31 11:17:21,153 org.apache.mesos.elasticsearch.systemtest.DiscoverySystemTest call - Polling Elasticsearch _nodes endpoints...
    ...
    [INFO] 2015-07-31 11:22:15,970 org.apache.mesos.elasticsearch.systemtest.DiscoverySystemTest call - Polling Elasticsearch _nodes endpoints...
    [INFO] 2015-07-31 11:22:17,001 org.apache.mesos.elasticsearch.systemtest.DiscoverySystemTest call - Polling Elasticsearch _nodes endpoints...
    [INFO] 2015-07-31 11:22:18,043 org.apache.mesos.elasticsearch.systemtest.DiscoverySystemTest call - Polling Elasticsearch _nodes endpoints...
    [INFO] 2015-07-31 11:22:18,084 org.apache.mesos.mini.MesosCluster stop - Stopping Mesos cluster
    [INFO] 2015-07-31 11:22:18,085 org.apache.mesos.mini.MesosCluster stop - Removing container [tinyproxy_-1409758626]
    [INFO] 2015-07-31 11:22:18,247 org.apache.mesos.mini.MesosCluster stop - Removing container [registry_-2087876814]
    [INFO] 2015-07-31 11:22:18,694 org.apache.mesos.mini.MesosCluster stop - Removing container [mini_mesos_cluster_-1261598898]
    [INFO] 2015-07-31 11:22:19,453 org.apache.mesos.mini.MesosCluster stop - Removing container [elasticsearch-scheduler_-1132058764]

org.apache.mesos.elasticsearch.systemtest.DiscoverySystemTest > testNodeDiscoveryRest FAILED
    com.jayway.awaitility.core.ConditionTimeoutException at DiscoverySystemTest.java:26
    com.github.dockerjava.api.NotFoundException
```

**Comment by jhftrifork** [comment]
The system test `testNodeDiscoveryRest` is testing whether, after launching a Mesos scheduler with three Elasticsearch tasks, the Elasticsearch endpoint eventually reports the presence of three Elasticsearch nodes (one Elasticsearch node per Mesos task).

The test is implemented by polling an HTTP endpoint 'http://172.17.0.67:9202/_nodes', where that IP is a Mesos master, which is supposed to be running Elasticsearch. It expects that, after less than 5 minutes, the response should be JSON which lists three Elasticsearch nodes.

The test fails because the HTTP response is always an HTTP 500, "Unable to connect". This response is coming from an instance of tiny proxy, because it cannot route the request.

**Comment by jhftrifork** [comment]
@philwinder and @lguminski say that this bug only appears when running docker-machine on OS X. It does not appear on Linux. It does not appear when running boot2docker on OS X.

I will try to confirm that this does not appear when using boot2docker instead of docker-machine.

**Comment by jhftrifork** [comment]
@philwinder @lguminski I can confirm that this bug does not appear when using boot2docker instead of docker-machine on OS X.

**Comment by jhftrifork** [comment]
@lguminski reports that this bug is reproducible even when using boot2docker on OS X. So it's not simply "docker-machine on OS X doesn't work".

**Comment by jhftrifork** [comment]
Our system tests use https://github.com/ContainerSolutions/mini-mesos as a development instance of Mesos. The problem seems like it would be some kind of networking issue in mini-mesos running on docker-machine on OS X. However the system tests for mini-mesos do not capture the problem, because they pass on my machine with both docker-machine and boot2docker.

**Comment by frankscholten** [comment]
Any updates on this one?

**Comment by frankscholten** [comment]
Fixed by https://github.com/mesos/elasticsearch/pull/205?

**Comment by philwinder** [comment]
Yes, fixed by #205. System tests work on OSX now.
